suite/spdz/src/test/java/dk/alexandra/fresco/suite/spdz/TestMaliciousBehaviour.java

```diff
@@ -52,73 +51,53 @@ public void reset() {
 
   }
 
-  @Test
-  public void testCommitmentCorruptRound1() {
+  @Test(expected=MaliciousException.class)
+  public void testCommitmentCorruptRound1() throws Throwable {
     try {
       runTest(new CompareTests.TestCompareEQ<>(), EvaluationStrategy.SEQUENTIAL_BATCHED, 3,
           Corrupt.COMMIT_ROUND_1);
-      Assert.fail("Should not go well");
     } catch (RuntimeException e) {
-      if (e.getCause().getCause() == null || !(e.getCause()
-          .getCause() instanceof MaliciousException)) {
-        Assert.fail();
-      }
+      throw e.getCause().getCause();
     }
   }
 
-  @Test
-  public void testOpenCommitmentCorruptRound1() {
+  @Test(expected=MaliciousException.class)
+  public void testOpenCommitmentCorruptRound1() throws Throwable {
     try {
       runTest(new CompareTests.TestCompareEQ<>(), EvaluationStrategy.SEQUENTIAL_BATCHED, 2,
           Corrupt.OPEN_COMMIT_ROUND_1);
-      Assert.fail("Should not go well");
     } catch (RuntimeException e) {
-      if (e.getCause().getCause() == null || !(e.getCause()
-          .getCause() instanceof MaliciousException)) {
-        Assert.fail();
-      }
+      throw e.getCause().getCause();
     }
   }
 
-  @Test
-  public void testCommitmentCorruptRound2() {
+  @Test(expected=MaliciousException.class)
+  public void testCommitmentCorruptRound2() throws Throwable {
     try {
       runTest(new CompareTests.TestCompareEQ<>(), EvaluationStrategy.SEQUENTIAL_BATCHED, 3,
           Corrupt.COMMIT_ROUND_2);
-      Assert.fail("Should not go well");
     } catch (RuntimeException e) {
-      if (e.getCause().getCause() == null || !(e.getCause()
-          .getCause() instanceof MaliciousException)) {
-        Assert.fail();
-      }
+      throw e.getCause().getCause();
     }
   }
 
-  @Test
-  public void testOpenCommitmentCorruptRound2() {
+  @Test(expected=MaliciousException.class)
+  public void testOpenCommitmentCorruptRound2() throws Throwable {
     try {
       runTest(new CompareTests.TestCompareEQ<>(), EvaluationStrategy.SEQUENTIAL_BATCHED, 2,
           Corrupt.OPEN_COMMIT_ROUND_2);
-      Assert.fail("Should not go well");
     } catch (RuntimeException e) {
-      if (e.getCause().getCause() == null || !(e.getCause()
-          .getCause() instanceof MaliciousException)) {
-        Assert.fail();
-      }
+      throw e.getCause().getCause();
     }
   }
 
-  @Test
-  public void testMaliciousInput() {
+  @Test(expected=MaliciousException.class)
+  public void testMaliciousInput() throws Throwable {
     try {
       runTest(new BasicArithmeticTests.TestInput<>(), EvaluationStrategy.SEQUENTIAL_BATCHED, 2,
           Corrupt.INPUT);
-      Assert.fail("Should not go well");
     } catch (RuntimeException e) {
-      if (e.getCause().getCause() == null || !(e.getCause()
```

Comment:
though prettier, does not have a the same sematic

Reply:
True, but the test will fail/pass on the same cases. And it is much more clear what we are testing for now.

Reply:
If the latter statement was true, then the emantic is identical. Which it is not. Call me for explanation.

As I said, it is prettier so this is just a comment. Do not change anything